src/tutorials/0004-mutable-file-system/05.md

```diff
@@ -23,20 +23,36 @@ such as `/catPics`,
 `files.ls` produces an array of objects, one for each file or directory
 contained in the directory you're inspecting, with the following properties:
 
-- `name` (default): the file's name
-- `type` (optional with `{ long: true }`): the object's type (`0` - file or `1` - directory)
-- `size` (optional with `{ long: true }`): the size of the file in bytes
-- `hash` (optional with `{ long: true }`): the cryptographic hash, or Content Identifier (CID), that represents your file in IPFS
-
-*Gotcha!* To return values for *all* of those properties, you must use the
-option `{ long: true }`. Otherwise all properties except for the `name` field will be
-present but their values will be returned as `0` (for `size` and `type`) or `""` (for `hash`).
+- `name`: the file's name
+- `type`: the object's type (`0` - file or `1` - directory)
+- `size`: the size of the file in bytes
+- `cid`: the Content Identifier (CID), that identifies your file in IPFS
 
 If we wanted to view all available details for the contents of a `/catPics`
 directory, we could do it like this:
 
 ```js
-await ipfs.files.ls('/catPics', { long: true })
+ipfs.files.ls('/catPics')
+```
+
+Because the `files.ls` method returns an [`Async Iterable`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for-await...of), you'll need to call the iterable multiple times until you have all the results.
```

Comment:
Hmm, you don't really call it, unless you're interfacing with the raw `.next()` method - you "iterate over it's values". This says you _need_ to get all the results, which I guess is true for this exercise as it stands, but you can stop iterating when you like, you don't have to consume it all. For example, if you're looking for a particular thing you can stop early, and actually that's better for the IPFS node because it doesn't need to fetch everything (from disk and/or the network).

Instead of the exercise being to collect items in an array why not do one of:

1. Return the async iterable
2. Log out the name and size of each file with `console.log` in the `for await` loop. You could easily stub `console.log`, collect the results and assert on what was logged when the function completes.

I think the latter is more interesting, especially since earlier we have "If we wanted to view all available details for the contents of a `/catPics`" - currently we don't "view" anything.

Reply:
The second option looks very interesting! 

@terichadbourne

Reply:
So after discussing this with @terichadbourne, there is a point to be made regarding the output that is shown to the user when they submit the correct solution code.

@alanshaw your proposals would create a difference behavior to the output: ATM _all_ output show the result returned from the solution code.
If we implement either (1) or (2) we would have to have change this behavior.
This inconsistency might cause some confusion to the user. We think it's very important to try and keep code solutions as close to real case scenarios as possible, and so, having the output remain consistent (return statements is what is visualized in the output) is crucial.

I really liked (2) though, but it's better to keep the UX consistent in this case.

Reply:
Understood - perhaps consider re-wording "If we wanted to view all available details for the contents of a `/catPics`" to not say "view"...? You'll get to "view" them because that's what protoschool does for you, but _you_ are not writing code to "view", you're just listing the contents and returning them.

Reply:
@alanshaw yeah good point.
Maybe we could re-word it to "inspect" the directory contents?